gateway/src/node/client_handling/websocket/connection_handler/fresh.rs

```diff
@@ -75,6 +81,10 @@ pub(crate) struct FreshHandler<R, S, St> {
     pub(crate) socket_connection: SocketStream<S>,
     pub(crate) storage: St,
     pub(crate) coconut_verifier: Arc<CoconutVerifier>,
+    // Occasionally the handler is requested to ping the connected client for confirm that it's
+    // active, such as when a duplicate connection is detected. This hashmap stores the oneshot
+    // senders that are used to return the result of the ping to the handler requesting the ping.
+    pub(crate) is_active_ping_pending_replies: HashMap<u64, IsActiveResultSender>,
```

Comment:
wait, I'm confused. why do we need an entire map for those? isn't it the case that we'd only be handling a single ping/pong at a time?  if we receive a message with wrong tag, we just ignore it.